### PR TITLE
Assignment 1 Alan Rose and Grant Howard

### PR DESCRIPTION
**Pull request by arose-13**
Assignment 1 - Alan Rose (@arose-13) and Grant Howard (@Howard404)

# Changes
* Null pointer exception was fixed
* Percentage is now calculated correctly
* 2 functions were added to shuffle the questions - extra credit
* Units tests were added to test each new function - extra credit

### CODE REVIEW

**Comment by arose-13** [comment]
Sorry about how our changes are showing up, we had to cherry-pick commits from another branch and I think that's why it shows that the whole content of each file was changed.

**Review by Nolan-Posey — COMMENTED** [review]
Pull Request title includes "Assignment 1" in the title ✔
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔
Issue 3: Add a new feature request implemented ✔
Issue 3: Feature request unit tested ✔

**Review by TriciaFaust — COMMENTED** [review]
Pull Request title includes "Assignment 1" in the title ✔
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔
Issue 3: Add a new feature request implemented ✔
Issue 3: Feature request unit tested ✔

I liked shuffling the questions, that's a good addition.

**Review by lsobtafo — COMMENTED** [review]
Pull Request title includes "Assignment 1" in the title ✔
Pull Request targets Assignment 1 branch ✔
Issue 1: Application no longer crashes ✔
Issue 2: Unit test properly passes ✔

File: PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs [hunk 1]
```diff
@@ -1,90 +1,134 @@
...
+
+            // Assert
+            Assert.IsTrue(File.Exists(filePath));
+        }
+
+        [TestMethod]
+        [DataRow(1, 1, "100%")]
+        [DataRow(5, 10, "50%")]
+        [DataRow(1, 10, "10%")]
+        [DataRow(0, 10, "0%")]
```
Comment (JLee35) [inline PrincessBrideTrivia/PrincessBrideTrivia.Tests/ProgramTests.cs:62]: Good testing, but we should always look for edge cases. An edge case to consider in this situation would be an odd percentage, say 3 out of 9 correct (we would expect 33% not 33.33%).

File: PrincessBrideTrivia/PrincessBrideTrivia/Program.cs [hunk 2]
```diff
@@ -1,96 +1,129 @@
...
+                {
+                    numberCorrect++;
+                }
+            }
+            Console.WriteLine("You got " + GetPercentCorrect(numberCorrect, questions.Length) + " correct");
+        }
+
+        public static string GetPercentCorrect(int numberCorrectAnswers, int numberOfQuestions)
+        {
+            decimal percentage = ((decimal)numberCorrectAnswers / numberOfQuestions * 100);
```
Comment (JLee35) [inline PrincessBrideTrivia/PrincessBrideTrivia/Program.cs:28]: Another edge case would be a `System.DivideByZeroException` here. It wouldn't be terrible to check this before attempting any arithmetic.
